src/Microsoft.AspNetCore.ResponseCaching/Internal/StringBuilderExtensions.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+
+namespace Microsoft.AspNetCore.ResponseCaching.Internal
+{
+    internal static class StringBuilderExtensions
+    {
+        internal static StringBuilder AppendUpperInvariant(this StringBuilder builder, string str)
+        {
+            for (var i = 0; i < str.Length; ++i)
+            {
+                builder.Append(char.ToUpperInvariant(str[i]));
```

Comment:
There was an in-place toupper but that is not safe and only works in very specific scenarios.